### PR TITLE
add rzil 'fexcept'|'FEXCEPT'

### PR DESCRIPTION
**Pull request by b1llow**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Add a new rzil op for getting floating point exceptions

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

**Review by wargio — APPROVED** [review]
this makes sense, but i think those should be handled like for div0. 

see `rz_il_handler_div`

**Review by XVilka — APPROVED** [review]

File: librz/il/il_opcodes.c [hunk 1]
```diff
@@ -945,6 +945,13 @@ RZ_API RZ_OWN RzILOpFloat *rz_il_op_new_frsqrt(RzFloatRMode rmode, RZ_NONNULL Rz
 	return ret;
 }
 
+RZ_API RZ_OWN RzILOpBool * rz_il_op_new_fexcept(RzFloatException e, RZ_NONNULL RzILOpFloat *x) {
```
Comment (XVilka) [inline librz/il/il_opcodes.c]:
```
librz/include/rz_il/rz_il_opcodes.h:769:27: error: code should be clang-formatted [-Wclang-format-violations]
RZ_API RZ_OWN RzILOpBool * rz_il_op_new_fexcept(RzFloatException e, RZ_NONNULL RzILOpFloat *x);
                          ^
```

**Review by XVilka — CHANGES_REQUESTED** [review]

**Review by XVilka — APPROVED** [review]

File: librz/il/theory_fbasic.c [hunk 2]
```diff
@@ -286,6 +286,38 @@ void *rz_il_handler_fsqrt(RzILVM *vm, RzILOpPure *op, RzILTypePure *type) {
...
+		break;
+	case RZ_FLOAT_E_OVERFLOW:
+		e = n->exception & RZ_FLOAT_E_OVERFLOW;
+		break;
+	case RZ_FLOAT_E_UNDERFLOW:
+		e = n->exception & RZ_FLOAT_E_UNDERFLOW;
+		break;
+	case RZ_FLOAT_E_INEXACT:
+		e = n->exception & RZ_FLOAT_E_INEXACT;
+		break;
```
Comment (wargio) [inline librz/il/theory_fbasic.c]: This should also generate an exception event like here: https://github.com/rizinorg/rizin/blob/e7b919f001c5894ee66aeae49567e6cf8dfc429a/librz/il/theory_bitv.c#L233

**Review by wargio — CHANGES_REQUESTED** [review]
see previous comment

**Review by XVilka — CHANGES_REQUESTED** [review]
1. Please add a test also
2. Please rebase to fix the CI

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Comment by XVilka** [comment]
@thestr4ng3r could you please give a quick look?

**Pull request merged**